### ISSUE TITLE
Messages and/or logic is confused about permissions/xattr/FAT

### ISSUE THREAD

**Issue by gdt** [bcpierce00/unison#638]
There are features that some but not filesystems have:
  * permissions
  * extended attributes
  * ACLs

Of these, not having permission is deficient, and I view the others as optional.

Sync should do something rational in all cases, and it should be automatic if possible.

Currently there is some "sync involving a FAT partition" option which seems to be code for "filesystem is deficient because it does not support permissions".

To fix, do one of
  * change the preference and any associated error messages to "filesystem does not support permissions".  Add a hint about that preference to the error message.
  * treat permissions like xattr and acl, and just don't try to set permissions on deficient filesystems.  Be careful not to result in permission changes on POSIX-compliant filesystems.  This is probably the better approach.

For context, see #591.   Note that this has not been reproduced on 2.51.5, and I'm making an exception (for a bug not reproduced on the most recent release).

**Comment by gdt** [comment]
OK, but is it really "FAT", or is it "deficient filesystem"?   And, it seems some variety of Windows filesystems have some of those abilities, and some don't.

**Comment by tleedjarv** [comment]
I don't think it is "deficient filesystem". NTFS for sure supports many-many things but those features might not be available on a non-Windows system. So, deficient driver, I'd wager.

Sure, it can't be _just_ FAT but what other widely-used fs does not support these basic features? (exFAT is also FAT, so doesn't count, in my view)

**Comment by gdt** [comment]
Good point about FS vs driver-and-FS.
